### PR TITLE
Divide penalty by n_batches

### LINKED ISSUE #213: Penalty (wrongly) not multiplied by the relative batch size
In the following line, I believe the penalty should be multiplied by the relative batch size (batch size / dataset size) such that the expectation of the stochastic gradient evaluated at the same parameter values is proportional to the true gradient.

https://github.com/FluxML/MLJFlux.jl/blob/4aae8c25df008aa5980f3031407c361628ddd6b0/src/core.jl#L39

You all probably know this already but here is the mathematical justification anyways. Let the full batch loss be `L(w)` for some parameter values `w` and let the mini-batch loss be the conditional random variable `l | w`. Additionally, let the batch size be `n` and the full dataset size be `N`. The expected value of `l | w` is:
```
E(l | w) = n/N * L(w)
```
Stochastic gradient descent relies on the fact that the expected value of the stochastic objective is proportional to the true full batch objective. Let the full batch objective be:
```
O(w) = L(w) + penalty(w)
```
Now let's consider the following 2 mini-batch objectives `o | w` and their expectations:
- `o | w = (l | w) + penalty(w)` -> `E(o | w) = E(l | w) + penalty(w) = n/N * L(w) + penalty(w)`
- `o | w = (l | w) + n/N * penalty(w)` -> `E(o | w) = E(l | w) + n/N * penalty(w) = n/N * L(w) + n/N * penalty(w) = n/N * O(w)`

The second mini-batch objective is the one whose expectation is proportional to the full batch objective `O(w)`. The first one over-penalises the weights by 1 over the relative batch size.

### PR DESCRIPTION
**Pull request by mohamed82008**
This PR fixes #213.

### CODE REVIEW

**Comment by mohamed82008** [comment]
This was easier than expected.

**Comment by ablaom** [comment]
That's great. I was thinking about a test. 

If our Flux model is pure *linear* (eg,`builder = MLJFlux.Linear(σ=identity, rng=123)`)  and we assume all batch sizes are equal, then isn't it true that training should be independent of batch size, with or without loss penalty?

**Comment by ablaom** [comment]
If we additionally assume the training target is constant, that is.
